CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/DelimitedImportController.java

```diff
@@ -75,9 +80,14 @@ public void setfilesIncludeHeaders(final boolean filesIncludeHeaders) {
         if (this.filesIncludeHeaders != filesIncludeHeaders) {
             this.filesIncludeHeaders = filesIncludeHeaders;
             updateSampleData();
+            validateFileStructure(files);
         }
     }
 
+    public List<File> getFiles() {
+        return files;
```

Comment:
Are you able to make this protected (since it's only being used for unit test). Can you also make it return an unmodifiable version of the list as well (``Collections.unmodifiableList...))?